### PR TITLE
HCAL TP Geometry changes for plan1, phase2

### PR DESCRIPTION
**Pull request by matz-e**
Add some bits to fully activate phase2, and add some placeholders to activate for plan1.

### CODE REVIEW

**Comment by rekovic** [comment]
Please test

**Comment by rekovic** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
@matz-e Looks like it needs rebase.

**Comment by ianna** [comment]
please test

**Comment by kpedro88** [comment]
needs another rebase...

**Comment by ianna** [comment]
+1

File: CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc [hunk 1]
```diff
@@ -204,12 +204,23 @@ CaloTPGTranscoderULUT::getOutputLUTSize(const HcalTrigTowerDetId& id) const
       case HcalTopologyMode::TriggerMode_2016:
          return QIE8_OUTPUT_LUT_SIZE;
       case HcalTopologyMode::TriggerMode_2017:
+         if (id.ietaAbs() <= theTopology->lastHERing())
+            return QIE8_OUTPUT_LUT_SIZE;
+         else
+            return QIE10_OUTPUT_LUT_SIZE;
+      case HcalTopologyMode::TriggerMode_2017plan1:
```
Comment (franzoni) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: hello @matz-e
the case 2017plan1 
is the baseline for CMS at this stage: are you planning on implementing it any time soon?
At this moment, 

case HcalTopologyMode::TriggerMode_2017plan1:

would proceed to execute the:

case HcalTopologyMode::TriggerMode_2018:

block, which is a worse approximation of TriggerMode_2017plan1 than TriggerMode_2017 is.

Comment (matz-e) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: Yes, I plan on getting the implementation done as soon as this is in and we fix the gain widths needed for the LUTs.

**Comment by franzoni** [comment]
+1

File: CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc (hunk 1 above)
Comment (davidlange6) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc]: hi @matz-e - for the next iteration - a small lookup table that lets you rewrite all this code as return lutSIze_[id.ietaAbs];  might be worth considering.

**Comment by davidlange6** [comment]
hi- @matz-e -looks like this is causing problems in the relvals. Is there a quick fix?

https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_9_0_X_2017-02-14-2300/pyRelValMatrixLogs/run/10826.0_QCD_Pt_600_800_13+QCD_Pt_600_800_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step2_QCD_Pt_600_800_13+QCD_Pt_600_800_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

**Comment by davidlange6** [comment]
Seems that there is not - will revert for now - just remake this PR with a fix. Thanks

**Comment by matz-e** [comment]
Sorry, just got in. I'll have a look at a fix and will remake it.

**Comment by matz-e** [comment]
@davidlange6 This is due to the geometry being used from the DB, and me redefining the enum values. I'll fix that with my other open PR, by keeping the meaning of the integers in the enum constant, and adjusting labels.

**Pull request merged**